### PR TITLE
Edit and delete messages

### PR DESCRIPTION
**Pull request by parenthetical**

### CODE REVIEW

**Comment by parenthetical** [comment]
I think this works correctly, how does the code look to you @stevekrouse?

The style and layout needs work still especially for mobile.

File: src/components/Messages.tsx [hunk 1]
```diff
@@ -34,19 +34,22 @@ export default function Messages({ channel }: { channel: Channel }) {
...
-                    }}
-                    key={index}
-                    message={message}
-                    channel={channel}
-                  />
-                )
+                ) =>
+                  message.deleted ? (
+                    <></>
+                  ) : (
```
Comment (stevekrouse) [inline src/components/Messages.tsx]: I'd probably filter out the deleted messages above, so we don't have to handle them here

**Comment by stevekrouse** [comment]
Code looks excellent!

**Comment by stevekrouse** [comment]
The demo works! From a UI perspective, I try to copy from Slack or Discord (or another chat platform), and what slack does is:

1. have a "more actions" menu show up on hover on the message
2. hide these actions under that menu
3. confirm the delete with a modal
4. Have a clearer "save changes" or "cancel edit" UI for editing
5. If a message has submessages and is deleted, the submessages stay alive and just show that their parent "has been deleted"

Here's what I think we should do:

1. Only show the actions on hover (this will require tailwindcss, so it can wait until I add that) --> https://github.com/compose-run/server/issues/44
2. Have the editing UI look more like Slack's (instead of reusing the existing top-level buttons like you're doing now)

(Skip the "more actions" menu because we only have 2 actions now; skip the delete confirm modal for now; our architecture already doesn't actually delete the messages, which is great for threads of deleted messages)

Sorry that was so long-winded, my main note for this PR is re-doing the edit functionality to be more like Slack:

Screen Shot 2021-12-13 at 10 50 20 AM

Oh, and one other point, the enter key should probably submit the edit changes (like it does for sending a message).

**Pull request merged**